myia/compile/transform.py

```diff
@@ -9,6 +9,7 @@
 def set_types(graph, argspec, outspec, pipeline):
     """Re-infer and set all the types for the nodes."""
     pipeline.resources.inferrer.infer(graph, argspec, outspec, clear=True)
+    graph.manager.keep_roots()
```

Comment:
Shouldn't this be done before inference?

Reply:
The extra graph pops up during the inference process. And come to think of it, I figured out why this happens: the new version of `switch` explicitly calls `engine.pipeline.resources.convert(bool)`, because the condition to `if` is usually wrapped with `bool`, and `switch` wants to skip over it to find the real operation. I will check if I can find a better way to do this, then that call won't be necessary anymore.

Reply:
Ok